### ISSUE TITLE
Shift the responsibility for calling `updateSessionTokenInBackground()`

### ISSUE THREAD

**Issue by philbooth** [mozilla/fxa-auth-server#1008]
From [this comment](https://github.com/mozilla/fxa-auth-server/pull/997#issuecomment-128749339):

> What do you think about pushing updateSessionTokenInBackground calls down a level? Like either calling it from db.sessionToken or even maybe a new db-server endpoint that does the get and update simultaneously? This get & touch is sort of a new pattern so I'm uncertain what the best way is. Either way it'll kind of bork up the current abstraction for fetching tokens so lets get this working code out first.

**Comment by philbooth** [comment]
> or even maybe a new db-server endpoint that does the get and update simultaneously?

This appealed to me initially because it reduces the number of requests, but on second thoughts it doesn't really fit with the RESTful API. So I think, at that level, the GET and the POST should remain separate (and it's not like the extra round-trip holds anything else up, since we always do it fire-and-forget anyway).

But inside this repo, I agree that explicitly making the same two function calls every time we want to get a session token is stupid. So I think a unified `DB.prototype.getAndUpdateSessionToken` in `lib/db.js` might be the correct thing to do.

**Comment by philbooth** [comment]
@dannycoates, in `hapi-auth-hawk`, is there a clever way to access the request object from inside a `getCredentialsFunc`?

Because `db.sessionToken` is always called from that context, `db.updateSessionTokenInBackground` must be callable from the same place if we are to combine the two. And as it takes the user-agent string for its second argument, I need access to the request object to call it.

I checked the docs and they only mention `id` and `callback` as arguments. And I dumped `this` at runtime just in case, but no. Is it possible?
